cms/search/publishers.py

```diff
@@ -0,0 +1,202 @@
+        # For pages that are 'release' content, we add release fields below.
+        content_type = self._map_page_type_to_content_type(page)
+
+        # Common fields for StandardPayload (also part of ReleasePayload)
+        # The schema requires at minimum: uri, title, content_type
+        message = {
+            "uri": page.url_path,
+            "content_type": page.content_type_id,
+            "release_date": (
+                page.release_date.isoformat().replace("+00:00", "Z")
+                if content_type == "release" and getattr(page, "release_date", None)
+                else None
+            ),
+            "summary": page.summary,
+            "title": page.title,
+            "topics": list(page.topics.values_list("topic_id", flat=True)) if hasattr(page, "topics") else [],
+        }
+
+        # If it's a Release, we add the extra fields from ReleasePayload
+        if content_type == "release":
+            message.update(self._construct_release_specific_fields(page))
+        return message
+
+    def _construct_release_specific_fields(self, page):
+        """Constructs and returns a dictionary with release-specific fields."""
+        release_fields = {
+            "finalised": page.status in ["CONFIRMED", "PROVISIONAL"],
+            "cancelled": page.status == "CANCELLED",
+            "published": page.status == "PUBLISHED",
+            "date_changes": [],
+        }
+
+        # If page.changes_to_release_date exists, add the first item's reason & previous date
+        if hasattr(page, "changes_to_release_date") and page.changes_to_release_date:
+            first_change = page.changes_to_release_date[0].value
+            release_fields["date_changes"].append(
+                {
+                    "change_notice": first_change.get("reason_for_change"),
+                    "previous_date": str(first_change.get("previous_date").isoformat().replace("+00:00", "Z"))
+                    if first_change.get("previous_date")
+                    else None,
+                }
+            )
+        return release_fields
+
+    def _construct_message_for_delete(self, page):
+        """Build a dict that matches the 'content-deleted' schema:
+        - required: uri
+        - optional: trace_id.
+        """
+        message = {
+            "uri": page.url_path,
+        }
+
+        if hasattr(page, "trace_id"):
+            message["trace_id"] = page.trace_id
+
+        return message
+
+    def _map_page_type_to_content_type(self, page):
+        """Maps the class name of a given page object to a corresponding content type string."""
+        page_class_name = page.__class__.__name__
+
+        mapping = {
+            "ReleaseCalendarPage": "release",
+            "StatisticalArticlePage": "bulletin",
+            "InformationPage": "static_page",
+            "IndexPage": "static_landing_page",
+            "MethodologyPage": "static_methodology",
+        }
+
+        return mapping.get(page_class_name)
```

Comment:
```suggestion
        return mapping.get(page.__class__.__name__)
```

without `page_class_name = page.__class__.__name__`

Reply:
Done!